FutureFeatures.md

```diff
@@ -179,6 +179,56 @@ should support:
 * Code unloading capabilities, especially in the context of code garbage
   collection and defragmentation.
 
+## Optimization and JIT library
+
+WebAssembly's JIT interface will likely be fairly low-level, exposing
+general-purpose primitives rather than higher-level functionality. Still,
+there is a need for higher-level functionality, and for greater flexibility
+than the WebAssembly spec can provide. An optimization and JIT library that
+would run inside WebAssembly and provide support and higher-level features
+would fit this need very well.
+
+Such a library wouldn't be part of the WebAssembly spec itself, but it's
+relevant to discuss here because features that we can expect to address in the
+library are features that we can avoid adding to the spec.
```

Comment:
I think this is good (we'd discussed in person, not written down yet), but should probably be at the top of this file, and referred to from here?